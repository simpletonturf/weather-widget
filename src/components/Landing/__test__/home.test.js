import React from "react";
import ReactDOM from "react-dom";
import Home from "./../home";
import DayCard from "../../weatherComponents/dayCard/dayCard";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
    const div=document.createElement("div");
    ReactDOM.render(<DayCard/>, div)
})

it("renders without crashing", () => {
    const div=document.createElement("div");
    ReactDOM.render(<Home/>, div)
})

it("renders card correctly", () =>{
    const {getByTestId}=render(<DayCard
        date="2020-11-10"
        icon="hr"
        maxtemp="23"
        mintemp="15"
        humid="75"
        system="metric" 
    />)
    expect(getByTestId("day_test")).toHaveTextContent("Today");
    expect(getByTestId("temp_humidity")).toHaveTextContent("75%");
})


it("renders card correctly", () =>{
    const {getByTestId}=render(<DayCard
        date="2020-11-11"
        icon="hr"
        maxtemp="24"
        mintemp="20"
        humid="82"
        system="imperial" 
    />)
    expect(getByTestId("day_test")).toHaveTextContent("Wednesday");
    expect(getByTestId("temp_humidity")).toHaveTextContent("82%");
})

it("matches snapshot", ()=>{
    const tree= renderer.create(<DayCard
        date="2020-11-10"
        icon="hr"
        maxtemp="23"
        mintemp="15"
        humid="75"
        system="metric" 
    />).toJSON();
    expect(tree).toMatchSnapshot();
})






