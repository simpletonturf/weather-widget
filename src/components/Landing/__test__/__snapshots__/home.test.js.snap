// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="daycard"
>
  <center>
    <h3
      data-testid="day_test"
    >
      Today
    </h3>
    <img
      alt="Weather Icon"
      src="https://www.metaweather.com/static/img/weather/png/64/hr.png"
    />
    <br />
    <strong
      data-testid="temp_test"
    >
      23
      ℃ 
      15
      ℃
    </strong>
    <br />
    <strong
      className="humid"
      data-testid="temp_humidity"
    >
      Humidity 
      75
      %
    </strong>
  </center>
</div>
`;
